### PR TITLE
Clear Log and Tab Log feature add

### PR DESCRIPTION
**Pull request by juggernautsei**
This feature add is to be able to clear and store the billing log and open the log in another tab so that it is easier to work claims.

### CODE REVIEW

**Comment by stephenwaite** [comment]
hi @juggernautsei , i added your repo as a remote and just checked out your clear_Log_feature and will test out

File: interface/billing/billing_report.php [hunk 1]
```diff
@@ -617,9 +619,13 @@ function MarkAsCleared(Type) {
                                            title='<?php echo xla('See messages from the last set of generated claims'); ?>'><strong><?php echo xlt('View Log'); ?></strong></a>
                                     </li>
                                 <?php } ?>
+                                <li><a href="<?php echo $webroot ?>/interface/billing/customize_log.php" target="_blank"><?php  echo xlt('Tab Log') ?></a>
```
Comment (stephenwaite) [inline interface/billing/billing_report.php]: tab log is missing strong styling like the other options

File: interface/billing/billing_report.php [hunk 2]
```diff
@@ -617,9 +619,13 @@ function MarkAsCleared(Type) {
...
+                                </li>
                                 <li><a class="link_submit"
                                        href="JavaScript:void(0);" onclick="select_all(); return false;"><strong><?php echo xlt('Select All'); ?></strong></a>
                                 </li>
+                                <li><a  id="clear-log" href="#" class='btn btn-default btn-sm' title='<?php xla('Clear the log'); ?>'><?php echo xlt('Clear Log') ?></a>
+                                </li>
```
Comment (stephenwaite) [inline interface/billing/billing_report.php]: button looks cool here but the others are button less unlike the screenshot https://community.open-emr.org/t/clearing-the-billing-log-feature/12018?u=stephenwaite

File: interface/billing/clear_log.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+     
+require_once("../globals.php");
+
+$filename = $GLOBALS['OE_SITE_DIR'] . '/edi/process_bills.log';
+$date = date("Y-m-d");
+
+$newlog = $GLOBALS['OE_SITE_DIR'] . '/edi/' . $date . '_process_bills.log';
```
Comment (stephenwaite) [inline interface/billing/clear_log.php]: the log is under the documents folder now so would need `/docments/edi/`

File: interface/billing/billing_report.php [hunk 4]
```diff
@@ -1361,6 +1367,14 @@ function MarkAsCleared(Type) {
         top.restoreSession();
         dlgopen('customize_log.php', '_blank', 500, 400);
     });
+    $("#clear-log").click( function(){
+        var checkstr = confirm('Do you really want to clear the log?');
```
Comment (bradymiller) [inline interface/billing/billing_report.php]: translate and escape for js:
```
var checkstr = confirm(<?php echo xlj("Do you really want to clear the log?"; ?>);
```

File: interface/billing/billing_report.php [hunk 5]
```diff
@@ -1361,6 +1367,14 @@ function MarkAsCleared(Type) {
...
+        if(checkstr == true){
+            dlgopen("clear_log.php", '_blank', 500, 400);
```
Comment (bradymiller) [inline interface/billing/billing_report.php]: Would it be better to just call this script via ajax?

Comment (bradymiller) [reply interface/billing/billing_report.php]: Disregard my post above; I see you are running this script as a popup.

**Comment by bradymiller** [comment]
btw, very nice code review @stephenwaite !!

File: interface/billing/billing_report.php (hunk 2 above)
Comment (juggernautsei) [reply interface/billing/billing_report.php]: Yes caught <li> styling change but did think to remove the button class.

Comment (stephenwaite) [reply interface/billing/billing_report.php]: clear log just needs a strong tag
was hoping you'd add the buttons :wink:

**Comment by juggernautsei** [comment]
@stephenwaite Can this be added to the next patch?

File: interface/billing/billing_report.php (hunk 2 above)
Comment (stephenwaite) [reply interface/billing/billing_report.php]: hi @juggernautsei, pretty sure @sjpadgett could cherry pick this once it's committed but think the clear log still needs a strong tag, thank you

Comment (juggernautsei) [reply interface/billing/billing_report.php]: I added the stong tag now. I tried making this into buttons but it just doesn't look right. Bootstrap makes the buttons to large even if I use sm tag. And styling the ul causes even more padding to be added by the browser. The only thing that would improve the look would be to remove the bullets.

**Comment by bradymiller** [comment]
@juggernautsei ,
I continued your work on this PR:
https://github.com/openemr/openemr/pull/2647/files#diff-815dba801e0d5ef540ecd6fcce88d63d

**Comment by juggernautsei** [comment]
Ok, what does that mean for me? Is it still in review?

**Comment by bradymiller** [comment]
hi @juggernautsei , Just brought it into the codebase (trying to keep the momentum going :) ). Thanks for the feature! -brady

**Comment by juggernautsei** [comment]
I know you have a lot to keep up with and I am just one squeaky wheel. 
On the horizon, I am looking into the GoodRx API/Widget and USA/Rx API. Both of these are drug savings programs that can be embedded in the system so at the point of prescribing they can be given the price of the drug and which pharmacy to get the drug at the discount price.  
Your thoughts.
The Weno thing did not work out for me in the respects of bringing in any revenue. It was an unfulfilled promise. My integrity won't let me abandon the project. To date, nearly 6000 prescriptions have been sent using the Weno portal. While not a huge number in the grand scheme of things. Some doctors out there depend on it. So, I keep building as the moto of OpenEMR goes.
I keep looking for avenues to keep the lights on here. GMU was a big help in that area. 
More stuff coming from my little corner.

**Pull request closed without merging**